pkg/network/dhcpconfigurator/dhcp_configurator.go

```diff
@@ -1,18 +1,43 @@
 package dhcpconfigurator
 
 import (
+	"fmt"
+	"os"
+
+	v1 "kubevirt.io/client-go/api/v1"
 	"kubevirt.io/kubevirt/pkg/network/cache"
+	netdriver "kubevirt.io/kubevirt/pkg/network/driver"
 )
 
 type DhcpConfigurator struct {
+	advertisingIfaceName     string
 	dhcpConfigurationFactory cache.InterfaceCacheFactory
+	filterByMac              bool
+	handler                  netdriver.NetworkHandler
 	launcherPID              string
 }
 
-func NewDhcpConfigurator(configurationFactory cache.InterfaceCacheFactory, launcherPID string) DhcpConfigurator {
+// NewDhcpConfiguratorWithClientFilter should be used when the DHCP server is
+// expected to only reply to the MAC specified in the `cache.DhcpConfig` struct
+func NewDhcpConfiguratorWithClientFilter(configurationFactory cache.InterfaceCacheFactory, launcherPID string, advertisingIfaceName string, handler netdriver.NetworkHandler) DhcpConfigurator {
```

Comment:
IIUC, the reason for this method is to avoid passing a boolean.
I suggest having something like enum, similar to [iptables.Protocol](https://github.com/AlonaKaplan/kubevirt/blob/d2aff45140ec776882385f680b46d8434bfa4021/vendor/github.com/coreos/go-iptables/iptables/iptables.go#L57).
It will keep the readability purpose and we will no longer need `NewDhcpConfiguratorWithClientFilter` and separate methods for `newBridgeDhcpConfigurator` and `newMasqueradeDhcpConfigurator` in `podInterface.go`.

Reply:
Enum instead of a boolean will improve the readability of the method invocation. Any method with an `if` does more than one thing, are we going to avoid `if`s as well?

Anyway, now two of the 4 methods were removed I won't insist on removing `NewDhcpConfiguratorWithClientFilter` as well.